ROADMAP.md

```diff
@@ -1,31 +1,42 @@
 # Roadmap
 
-This document defines a high level roadmap for Rook development. The dates below are subject to change but should give a general idea of what we are planning. We use the [milestone](https://github.com/rook/rook/milestones) feature in Github so look there for the most up-to-date and issue plan.
+This document defines a high level roadmap for Rook development.
+The dates below are subject to change but should give a general idea of what we are planning.
+We use the [milestone](https://github.com/rook/rook/milestones) feature in Github so look there for the most up-to-date and issue plan.
 
-## Rook 0.7
+## Rook 0.8
 
-- Ceph Block Storage (Cluster and Pool CRDs) declared beta
-- Mon reliability (restarts, failing over too fast, ip changes, etc.)
-- Durability of state (local storage support, config is regenerat-able)
-- Run everywhere (rook-agent uses ceph-fuse, nbd-rbd / tcmu runner)
-- Adding / removing nodes and disk drives (lifecycle issues, failures, etc.)
-- Run with Least Privileged and possibly without privileged containers
-- Shutdown / restart issues
-- Improved data placement and pool configuration (CRUSH maps)
-- Placement group balancer support (ceph-mgr module)
-- Use new Prometheus ceph-mgr module
-- Performance Testing for Block storage
-- Make API and CLI Optional (plan to remove)
-- Dynamic Volume Provisioning for CephFS
+* Multiple storage backends
+  * Full design
+  * Refactor code base and repositories to enable
+  * Consider support for Minio, potentially early support for other backends time permitting
+* Switch CRDs to API Aggregation
+* Run on arbitrary PVs
+* Remove Rook API and CLI
+* Migrate CI and release pipelines to a solution hosted by the CNCF
+* Run with Least Privileged and possibly without privileged containers
+* Shutdown / restart issues
+* Support Kubernetes 1.7+ only
+* Ceph features and improvements
+  * Adding/removing disks (lifecycle issues, failures, 1 OSD per pod, etc.)
+  * Placement group balancer support (ceph-mgr module)
+  * Mon reliability (restarts, failing over too fast, ip changes, etc.)
+  * Mimic support
 
-## Rook 0.8
+## Rook 0.9
+
+* Automated upgrade
+* Operator runs HA
```

Comment:
What does running the Operator in HA give us? Currently it is used to provision PV, and run health checks (and other things i cant remember) but with the controller-loop strategy of Kubernetes, these should not be an issue. IE, a pvc will be `pending` if the operator is unavailable. But once it is available, the PV would be provisioned.

Or is this done to minimize that window of unavailability?

Reply:
Minimizing the window of unavailability was probably the goal here.  But this specific effort likely isn't needed anymore once we go to API aggregation: https://github.com/rook/rook/issues/701#issuecomment-369349965

I'll remove it.